bcipy/helpers/triggers.py

```diff
@@ -638,3 +639,185 @@ def read_triggers(triggers_file: TextIO) -> List[Tuple[str, str, float]]:
+
+
+class TriggerHandler:
+
+    def __init__(self,
+                 path: str,
+                 file_name: str,
+                 flush_sens: FlushSensitivity):
+        self.path = path
+        self.file_name = f'{file_name}.txt'
+        self.flush_sens = flush_sens
+        self.triggers = []
+
+        if os.path.exists(self.file_name):
+            raise Exception(f"[{self.file_name}] already exists, any writing "
+                            "will overwrite data in the existing file.")
+
+        self.file = open(self.file_name, 'w+')
+
+    def close(self):
+        self.write()
+        self.file.close()
+
+    def write(self):
+        """
+        Writes current Triggers in self.triggers[] to .txt file in self.file_name.
+        File writes in the format "label, targetness, time".
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+            None
+        """
+
+        for trigger in self.triggers:
+            self.file.write(f'{trigger.label} {trigger.type.value} {trigger.time}\n')
+
+        self.triggers = []
+
+    @staticmethod
+    def read_text_file(path: str) -> List[str]:
+        if not path.endswith('.txt') or not os.path.exists(path):
+            raise FileNotFoundError(f"Valid triggers .txt file not found at [{path}]."
+                                    "\nPlease rerun program.")
+
+        triggers_list = []
+        with open(path) as raw_txt:
+            for line in raw_txt:
+                line_split = line.split()
+                triggers_list.append(line_split)
+        return triggers_list
+
+    @staticmethod
+    def load(path: str,
+             offset: Optional[float] = None,
+             exclusion: Optional[List[TriggerType]] = None) -> List[Trigger]:
+        """
+        Loads a list of triggers from a .txt of triggers.
+
+        Exclusion based on type only (ex. exclusion=[TriggerType.Fixation])
+
+        1. Checks if .txt file exists at path
+        2. Loads the triggers data as a list of lists
+        3. If offset provided, adds it to the time as float
+        4. If exclusion provided, filters those triggers
+        5. Casts all loaded and modified triggers to Trigger
+        6. Returns as a List[Triggers]
+
+        Parameters
+        ----------
+        path (str): name or file path of .txt trigger file to be loaded.
+            Input string must include file extension (.txt).
+        offset (Optional float): if desired, time offset for all loaded triggers,
+            positive number for adding time, negative number for subtracting time.
+        exclusion (Optional List[TriggerType]): if desired, list of TriggerTypes
+            to be removed from the loaded trigger list.
+
+        Returns
+        -------
+            List of Triggers from loaded .txt file with desired modifications
+        """
+
+        # Checking for file with given path, with or without .txt
+        triggers_list = TriggerHandler.read_text_file(path)
+
```

Comment:
There are some aspects of the implementation here that I'm finding confusing. In the type signature we state that offset is either a `float` or `None`, so it's not clear why we are checking to see if it's a `list`. Maybe the intention is that if an offset is provided then we want to exclude the last entry which has a `TriggerType` of 'offset_correction', in which case we will need an additional value added to the `TriggerType` enum.

I think the more common scenario is that rather than applying a manually provided offset we want the option of using the offset recorded in the trigger file. The offset parameter would then be a `bool`, where a True value applies this offset and a False value leaves the trigger times unchanged.

Reply:
If I'm understanding correctly, are you referring to these few lines in the `load` method?
```
if offset:
            if isinstance(offset, list) and exclusion is None:
                exclusion = offset
            else:
                for item in triggers_list:
                    time_float = float(item[2]) + offset
                    item[2] = str(time_float)
```
If you are, the reason I ended up having to do it like this was because if the `load` method was given an `exclusion` but no `offset`, the method would try to read in the `exclusion` as an `offset` due to the ordering of the parameters, which would fail and throw an error, since it was expecting a `float` but receiving a `list`. I couldn't figure out how to ignore the `offset` if it was `None` while an `exclusion` was present, so I ended up having to just do it like this.

In regards to the latter scenario you mentioned, I was unaware that this is how the offset is usually handled-- I can definitely change the handler and add a `TriggerCategory Enum` to fit this if needed.

Reply:
It seems like if we are going to use an offset it will always be the one provided in the triggers.txt file that we're reading, so I'm not sure it makes sense to have that as a parameter. You can check out the `read_triggers` function to see how this is handled (basically, we read all records in the file and `pop()` off the last result).  @sci-tab, do you have thoughts on this?

Reply:
We may have an offset or offsets provided in the triggers.txt... We should support loading / applying those optionally here. We also apply static offsets which are factored in outside of the triggers.txt file. @theJokerEvoker next time we meet let's discuss how to address this.

Reply:
@lawhead Tab and I have talked about it and made some adjustments to this issue-- we've made it so that the `read_text_file()` method now expects the first line of the `.txt` to be a Trigger that contains a "system offset" value, which will then be eventually returned to `load()` to be used. This assumption about the "system offset" will be implemented in [this ticket](https://www.pivotaltracker.com/story/show/179956988). We still maintain the ability to include a manual offset with the `load()`, however, though it is by default 0. Would you mind taking another look at the changes?

Reply:
@theJokerEvoker, thanks for your work on this. This version is a lot clearer for me.